crates/pallet-subspace/src/lib.rs

```diff
@@ -843,7 +843,7 @@ impl<T: Config> frame_support::traits::FindAuthor<T::AccountId> for Pallet<T> {
         digests
             .into_iter()
             .find_map(|(id, data)| DigestItemRef::PreRuntime(&id, data).as_subspace_pre_digest())
-            .map(|pre_digest| pre_digest.solution.public_key)
+            .map(|pre_digest| pre_digest.solution.reward_address)
```

Comment:
Is there a good place to put it (I had an idea to put it into `pallet-subspace`, but it would require compiling the whole pallet in order to use this trait in other pallets)?

Reply:
Actually, this `FindAuthor` trait is used both for fee collection and block rewards.

Reply:
Yeah, I'm not sure we want to change `FindAuthor` notion. This is just for rewards, block author (identity that signed it tied to the plot) is a different thing and can be punished, like plot burning. I also think it is important to not mix those together.

Reply:
>Is there a good place to put it (I had an idea to put it into pallet-subspace, but it would require compiling the whole pallet in order to use this trait in other pallets)?

`subspace-core-primitives` for now? Once we have more kinds of this stuff, a crate `subspace-pallet-support` might be useful. Perhaps Nazar could suggest a better place :P

Reply:
`subspace-core-primitives` is for substrate-independent things. This should probably go into `subspace-runtime-primitives`.